gammapy/cube/psf_map.py

```diff
@@ -331,3 +332,45 @@ def stack(self, other):
     def copy(self):
         """Copy PSFMap"""
         return deepcopy(self)
+
+    def sample_coord(self, map_coord, random_state=0):
+        """ Apply PSF corrections on the coordinates of a set of simulated events.
+
+        Parameters
+        ----------
+        map_coord : `~gammapy.maps.MapCoord` object.
+            Sequence of coordinates and energies of sampled events.
+        random_state : {int, 'random-seed', 'global-rng', `~numpy.random.RandomState`}
+            Defines random number generator initialisation.
+            Passed to `~gammapy.utils.random.get_random_state`.
+
+        Returns
+        -------
+        corr_coord : `~gammapy.maps.MapCoord` object.
+            Sequence of PSF-corrected coordinates of the input map_coord map.
+        """
+
+        random_state = get_random_state(random_state)
+
+        coord = {
+            "lon": map_coord.lon.reshape(-1, 1),
+            "lat": map_coord.lat.reshape(-1, 1),
+            "energy": map_coord["energy"].reshape(-1, 1),
+            "theta": (self.psf_map.geom.axes[0].center),
+        }
+
+        pdf = self.psf_map.interp_by_coord(coord)
```

Comment:
You're still creating a `coord` dict dropping frame information, and then passing it to `self.psf_map.interp_by_coord(coord)`, which can't work if `self.psf_map` and `map_coord` have a different frame.

Please:

1. Change your test case to be like that, i.e. fail because of the frame handling bug
2. Fix the code. Probably this works? `coord = {"skycoord": mapcoord.skycoord,...}`, or something similar.